quality-assurance/core.md

```diff
@@ -0,0 +1,192 @@
+* All errors are handled and UI reacts appropriately
+* Errors are clear in meaning &amp; reflect what happened
+* Activity indicators are used when loading data asynchronously
+* Empty information is handled correctly in UI (optional fields etc)
+* Long running operations give a indication of progress if possible
+* Breadcrumbs are used where appropriate
+
+**Navbar**
+* UI reflects the current active page
+* An icon that is appropriate for the page (if applicable)
+
+**Forms**
+* Invalid inputs correctly trigger validation errors
+* All form inputs have associated labels
+* Appropriate validation for different inputs (phone number, postcode)
+* The forms are not too long &amp; and are grouped logically
+* The user is alerted of the form lifecycle (success, error)
+* When modals are used in the context of updates, ensure inputs are correctly pre populated.
+* Autofill is disabled on inputs where appropriate
+* Inputs have correct type property (emails have email type, phone numbers have tel type)
+
+**Modals**
+* Modals are animated on both display and dismiss
+* Modals are used where appropriate over pages
+
+**Formatting**
+* Dates are formatted correctly for the appropriate audience.
+* Monetary values are formatted correctly for the appropriate audience
+* Phone numbers are formatted correctly for the appropriate audience
+
+**Pagination**
+* UI correctly indicates current page
+* UI shows at least 2 page tabs ahead and below the current page
+
+**Login &amp; Auth**
+* Users can only access pages they are permitted access to.
+* Authentication when they are logged in
+* Login, registration (if applicable) and logout works as expected.
+* Reset password works as expected
+
+**Registration**
+* Email addresses are verified &amp; unique (if applicable)
+* Passwords meet minimum standards (min 8 chars)
+* Passwords should not have maximum length restrictions (allow at least 255 characters).
+
+**Emails**
+* Emails look good on small form factor devices
+* Ensure correct subject &amp; sender name
+* Correct email is sent, information is conveyed appropriately
+* Emails are received in a timely manner
+* Emails come from the appropriate email (often the clients)
+* Links point to the correct URIs
+
+**Push Notifications**
+* Push notifications are received in a timely manner
+* They show correct &amp; relevant information
+
+**Permissions Roles**
+* Ensure permissions &amp; roles match up to possible actions (only admins can create users etc)
+
+### Whitebox
+**General**
+* Filenames must follow standard coding conventions (UserController.php)
+* Classes must follow standard coding conventions (UserController)
+* Variables should follow camelCase &amp; be descriptive (myName)
+* All long running operations (at your discretion) are queued
+* APIs follow REST architecture as closely as possible and implement our [API Response](https://github.com/langleyfoxall/helpers-laravel#apiresponse)
```

Comment:
```suggestion
* APIs follow REST architecture as closely as possible and implement our [API Response](https://github.com/langleyfoxall/helpers-laravel#apiresponse)
* API responses hide sensitive information outside of what is absolutely necessary.
```

Thoughts on this?

Reply:
I was more referring to the Laravel helpers section, but you didn't write that so it was more a comment on the actual PR 😄